### PR TITLE
[1.15.x] Fix modproperties in mods.toml causing exception

### PR DESCRIPTION
**Pull request by sciwhiz12**
_This PR is a LTS backport of #7192._

See the main PR for details of the issue and the fix.
_TL;DR:_ `modproperties` property in `mods.toml` allows custom data. [A commit][forgecommit] during 1.15 made use of the property throw a `ClassCastException`. Only reason crash was discovered was because someone [asked for help on Discord][discordconvo], which led to the root cause of the commit. This PR adds the missing `Optional`-chained call to `UnmodifiableConfig#valueMap`.

[forgecommit]: https://github.com/MinecraftForge/MinecraftForge/commit/20f78ac724164110d80c30a19473ee0bc1d2ecc1
[discordconvo]: https://discordapp.com/channels/313125603924639766/725850371834118214/738138530445918298

### CODE REVIEW

**Comment by sciwhiz12** [comment]
Updated PR to match the fix for 1.16.

**Comment by sciwhiz12** [comment]
Updated PR to match the 1.16 PR, fixing the `properties` property in `ModFileInfo`.

**Comment by TheCurle** [comment]
The 1.16 PR was merged. This message just serves as a reminder to those checking the github channel on Forgecord that this exists and is a useful feature.

**Comment by TheCurle** [comment]
For whatever strange reason, #7524 was created and merged before this PR was. Closing this as its intended purpose is fulfilled
